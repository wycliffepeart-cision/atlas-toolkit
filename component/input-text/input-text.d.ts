export declare const INPUT_TEXT_ATTRIBUTES: {
    DISABLED: string;
    REQUIRED: string;
    READONLY: string;
    NAME: string;
    VALUE: string;
    MAX_LENGTH: string;
    PATTERN: string;
    PLACEHOLDER: string;
    STATE: string;
};
export declare class InputText extends HTMLElement {
    #private;
    static template: any;
    static get observedAttributes(): string[];
    attributeChangedCallback(name: any, _oldValue: any, newValue: any): void;
    connectedCallback(): void;
    disconnectedCallback(): void;
    get value(): string;
    set value(value: string);
    get disabled(): boolean;
    set disabled(value: boolean);
    checkValidity(): boolean;
    get valid(): boolean;
    set valid(value: boolean);
}
//# sourceMappingURL=input-text.d.ts.map