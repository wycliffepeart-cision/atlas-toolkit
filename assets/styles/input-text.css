@layer atlas {
	atlas-input-text {
		--border-color: var(--atlas-color-primary-n100);
		--background-color: var(--atlas-input-common-background-color);
		--disabled-background-color: var(--atlas-color-primary-n50);
		--invalid-background-color: var(--atlas-color-semantic-danger-light);
		--invalid-border-color: var(--atlas-color-semantic-danger);
		--text-color: var(--atlas-color-primary-n400);
		--label-color: var(--atlas-color-primary-n300);

		--common-block-margin: 0.5em;
		--common-padding: 1em;
		--common-padding-inline: 3em;
		--input-padding-block-start: 1.7em;
		--input-padding-block-end: 0.7em;
		--label-padding-block-start: 0.6em;
		--affix-padding: 1.6em;
		--default-font-size: 1em;
		--smaller-font-size: 0.75em;
		--label-gap: 0.3em;

		--moz-label-padding-block: 0.2em;
	}

	atlas-input-text {
		accent-color: var(--atlas-input-common-accent-color);
		color: var(--text-color);
		display: grid;
		grid-template-areas: "Prefix Label Suffix" "Subtext Subtext Subtext";
		grid-template-columns: min-content 1fr min-content min-content;
		grid-template-rows: min-content min-content;
		margin-block: var(--common-block-margin);
	}

	atlas-input-text::part(input) {
		background-color: var(--background-color);
		border-color: var(--border-color);
		border-radius: var(--atlas-input-common-border-radius);
		border: var(--atlas-input-common-border);
		color: var(--text-color);
		font-size: var(--default-font-size);
		grid-row: 1;
		grid-column: 1 / span 4;
		padding-block: var(--common-padding);
		padding-inline: var(--common-padding-inline);
	}

	atlas-input-text[disabled]:not([disabled="false"])::part(input) {
		--border-color: var(--disabled-background-color);
		--background-color: var(--disabled-background-color);

		pointer-events: none;
		cursor: auto;
	}

	atlas-input-text::part(input):focus {
		box-shadow: var(--atlas-input-common-box-shadow);
		outline: none;
	}

	atlas-input-text::part(label-wrapper) {
		grid-area: Label;

		position: absolute;
		isolation: isolate;

		color: var(--label-color);
		display: flex;
		gap: var(--label-gap);

		padding-inline: var(--common-padding-inline);
		padding-block-start: var(--moz-label-padding-block);
	}

	atlas-input-text::part(label) {
		font-size: var(--smaller-font-size);
	}

	atlas-input-text[required]:not([required="false"])::part(required) {
		display: inline;
	}

	atlas-input-text::part(required) {
		display: none;
		vertical-align: super;
		font-size: smaller;
	}

	atlas-input-text::part(prefix),
	atlas-input-text::part(suffix) {
		font-size: var(--smaller-font-size);
		align-content: center;
		display: flex;
		flex-wrap: wrap;
	}

	atlas-input-text::part(prefix) {
		grid-area: Prefix;
		padding-inline-start: var(--affix-padding);
	}

	atlas-input-text::part(suffix) {
		grid-area: Suffix;
		padding-inline-end: var(--affix-padding);
	}

	atlas-input-text::part(helper-text) {
		grid-area: Subtext;

		display: none;
		font-size: var(--smaller-font-size);
	}

	atlas-input-text::part(validation-text) {
		grid-area: Subtext;

		color: var(--invalid-border-color);
		display: none;
		font-size: var(--smaller-font-size);
	}

	/* atlas--invalid CUSTOM STATE */
	atlas-input-text[atlas--invalid]::part(input) {
		--border-color: var(--invalid-border-color);
		--background-color: var(--invalid-background-color);
	}

	atlas-input-text:not([atlas--invalid])::part(helper-text) {
		display: block;
		margin-block-start: var(--common-block-margin);
	}

	atlas-input-text[atlas--invalid]::part(validation-text) {
		display: block;
		margin-block-start: var(--common-block-margin);
	}

	/* EXPERIMENTAL USE OF :has() */
	atlas-input-text:not(:has([slot="prefix"]))::part(input),
	atlas-input-text:not(:has([slot="prefix"]))::part(label-wrapper) {
		padding-inline-start: var(--common-padding);
	}

	atlas-input-text:not(:has([slot="suffix"]))::part(input) {
		padding-inline-end: var(--common-padding);
	}

	atlas-input-text:has([slot="label"])::part(input) {
		padding-block-start: var(--input-padding-block-start);
		padding-block-end: var(--input-padding-block-end);
	}

	atlas-input-text:has([slot="label"])::part(label-wrapper) {
		padding-block-start: var(--label-padding-block-start);
	}
}
