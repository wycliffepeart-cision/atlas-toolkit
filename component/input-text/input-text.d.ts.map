{"version":3,"file":"input-text.d.ts","sourceRoot":"","sources":["../../../src/component/input-text/input-text.ts"],"names":[],"mappings":"AAGA,eAAO,MAAM,qBAAqB;;;;;;;;;;CAUjC,CAAC;AAMF,qBACa,SAAU,SAAQ,WAAW;;IACxC,MAAM,CAAC,QAAQ,MAAC;IAEhB,MAAM,KAAK,kBAAkB,aAE5B;IAED,wBAAwB,CAAC,IAAI,KAAA,EAAE,SAAS,KAAA,EAAE,QAAQ,KAAA;IAIlD,iBAAiB;IASjB,oBAAoB;IAgCpB,IAAI,KAAK,WAER;IAED,IAAI,KAAK,CAAC,KAAK,QAAA,EAEd;IAED,IAAI,QAAQ,YAGX;IAED,IAAI,QAAQ,CAAC,KAAK,SAAA,EAEjB;IAED,aAAa;IAMb,IAAI,KAAK,YAER;IAED,IAAI,KAAK,CAAC,KAAK,SAAA,EAKd;CACF"}