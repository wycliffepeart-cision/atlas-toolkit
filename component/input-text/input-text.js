import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var InputTextTemplate = "<template>\n  <input id=\"Input\" type=\"text\" part=\"input\">\n\n  <label for=\"Input\" part=\"label-wrapper\">\n    <slot name=\"label\" part=\"label\"></slot>\n    <slot name=\"required\" part=\"required\">*</slot>\n  </label>\n\n  <slot name=\"prefix\" part=\"prefix\"></slot>\n  <slot name=\"suffix\" part=\"suffix\"></slot>\n\n  <slot name=\"helperText\" part=\"helper-text\"></slot>\n  <slot name=\"validationText\" part=\"validation-text\"></slot>\n</template>\n";

const INPUT_TEXT_ATTRIBUTES = {
    DISABLED: 'disabled',
    REQUIRED: 'required',
    READONLY: 'readonly',
    NAME: 'name',
    VALUE: 'value',
    MAX_LENGTH: 'maxlength',
    PATTERN: 'pattern',
    PLACEHOLDER: 'placeholder',
    STATE: 'state',
};
const INPUT_STATES = {
    ATLAS_INVALID: 'atlas--invalid',
};
let InputText = class InputText extends HTMLElement {
    static template;
    static get observedAttributes() {
        return Object.values(INPUT_TEXT_ATTRIBUTES);
    }
    attributeChangedCallback(name, _oldValue, newValue) {
        this.#attributePassthrought(name, newValue, [INPUT_TEXT_ATTRIBUTES.STATE]);
    }
    connectedCallback() {
        if (!this.isConnected) {
            return;
        }
        this.#inputElem.addEventListener('change', this.#handleInputChange);
        this.#inputElem.addEventListener('invalid', this.#handleInputInvalid);
    }
    disconnectedCallback() {
        this.#inputElem.removeEventListener('change', this.#handleInputChange);
        this.#inputElem.removeEventListener('invalid', this.#handleInputInvalid);
    }
    #attributePassthrought(name, value, exclude = []) {
        if (exclude.includes(name)) {
            return;
        }
        if (value === null) {
            this.#inputElem.removeAttribute(name);
        }
        else {
            this.#inputElem.setAttribute(name, value);
        }
    }
    #handleInputChange = ({ target }) => {
        this.dispatchEvent(new CustomEvent('change', { bubbles: true, composed: true }));
        const isValid = target.checkValidity();
        this.toggleAttribute(INPUT_STATES.ATLAS_INVALID, !isValid);
    };
    #handleInputInvalid = () => {
        this.dispatchEvent(new CustomEvent('invalid', { bubbles: true, composed: true }));
    };
    get #inputElem() {
        return this.shadowRoot.getElementById('Input');
    }
    get value() {
        return this.#inputElem.value;
    }
    set value(value) {
        this.#inputElem.value = value;
    }
    get disabled() {
        const disabledAttr = this.getAttribute('disabled');
        return disabledAttr !== null && disabledAttr !== 'false';
    }
    set disabled(value) {
        this.toggleAttribute('disabled', value);
    }
    checkValidity() {
        const isValid = this.#inputElem.checkValidity();
        this.toggleAttribute(INPUT_STATES.ATLAS_INVALID, !isValid);
        return isValid;
    }
    get valid() {
        return this.#inputElem.validity.valid;
    }
    set valid(value) {
        this.#inputElem.setCustomValidity(value ? '' : 'invalid');
        this.#inputElem.checkValidity();
        this.toggleAttribute(INPUT_STATES.ATLAS_INVALID, !value);
    }
};
InputText = __decorate([
    Template(InputTextTemplate)
], InputText);
if (!customElements.get('atlas-input-text'))
    customElements.define('atlas-input-text', InputText);

export { INPUT_TEXT_ATTRIBUTES, InputText };
