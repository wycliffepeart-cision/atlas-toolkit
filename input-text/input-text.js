import { _ as __decorate, a as __classPrivateFieldGet } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var InputTextTemplate = "<template>\n  <input id=\"Input\" type=\"text\" part=\"input\">\n\n  <label for=\"Input\" part=\"label-wrapper\">\n    <slot name=\"label\" part=\"label\"></slot>\n    <slot name=\"required\" part=\"required\">*</slot>\n  </label>\n\n  <slot name=\"prefix\" part=\"prefix\"></slot>\n  <slot name=\"suffix\" part=\"suffix\"></slot>\n\n  <slot name=\"helperText\" part=\"helper-text\"></slot>\n  <slot name=\"validationText\" part=\"validation-text\"></slot>\n</template>\n";

var _InputText_instances, _InputText_attributePassthrought, _InputText_handleInputChange, _InputText_handleInputInvalid, _InputText_inputElem_get;
const INPUT_TEXT_ATTRIBUTES = {
    DISABLED: 'disabled',
    REQUIRED: 'required',
    READONLY: 'readonly',
    NAME: 'name',
    VALUE: 'value',
    MAX_LENGTH: 'maxlength',
    PATTERN: 'pattern',
    PLACEHOLDER: 'placeholder',
    STATE: 'state',
};
const INPUT_STATES = {
    ATLAS_INVALID: 'atlas--invalid',
};
let InputText = class InputText extends HTMLElement {
    constructor() {
        super(...arguments);
        _InputText_instances.add(this);
        _InputText_handleInputChange.set(this, ({ target }) => {
            this.dispatchEvent(new CustomEvent('change', { bubbles: true, composed: true }));
            const isValid = target.checkValidity();
            this.toggleAttribute(INPUT_STATES.ATLAS_INVALID, !isValid);
        });
        _InputText_handleInputInvalid.set(this, () => {
            this.dispatchEvent(new CustomEvent('invalid', { bubbles: true, composed: true }));
        });
    }
    static get observedAttributes() {
        return Object.values(INPUT_TEXT_ATTRIBUTES);
    }
    attributeChangedCallback(name, _oldValue, newValue) {
        __classPrivateFieldGet(this, _InputText_instances, "m", _InputText_attributePassthrought).call(this, name, newValue, [INPUT_TEXT_ATTRIBUTES.STATE]);
    }
    connectedCallback() {
        if (!this.isConnected) {
            return;
        }
        __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).addEventListener('change', __classPrivateFieldGet(this, _InputText_handleInputChange, "f"));
        __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).addEventListener('invalid', __classPrivateFieldGet(this, _InputText_handleInputInvalid, "f"));
    }
    disconnectedCallback() {
        __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).removeEventListener('change', __classPrivateFieldGet(this, _InputText_handleInputChange, "f"));
        __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).removeEventListener('invalid', __classPrivateFieldGet(this, _InputText_handleInputInvalid, "f"));
    }
    get value() {
        return __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).value;
    }
    set value(value) {
        __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).value = value;
    }
    get disabled() {
        const disabledAttr = this.getAttribute('disabled');
        return disabledAttr !== null && disabledAttr !== 'false';
    }
    set disabled(value) {
        this.toggleAttribute('disabled', value);
    }
    checkValidity() {
        const isValid = __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).checkValidity();
        this.toggleAttribute(INPUT_STATES.ATLAS_INVALID, !isValid);
        return isValid;
    }
    get valid() {
        return __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).validity.valid;
    }
    set valid(value) {
        __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).setCustomValidity(value ? '' : 'invalid');
        __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).checkValidity();
        this.toggleAttribute(INPUT_STATES.ATLAS_INVALID, !value);
    }
};
_InputText_handleInputChange = new WeakMap(), _InputText_handleInputInvalid = new WeakMap(), _InputText_instances = new WeakSet(), _InputText_attributePassthrought = function _InputText_attributePassthrought(name, value, exclude = []) {
    if (exclude.includes(name)) {
        return;
    }
    if (value === null) {
        __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).removeAttribute(name);
    }
    else {
        __classPrivateFieldGet(this, _InputText_instances, "a", _InputText_inputElem_get).setAttribute(name, value);
    }
}, _InputText_inputElem_get = function _InputText_inputElem_get() {
    return this.shadowRoot.getElementById('Input');
};
InputText = __decorate([
    Template(InputTextTemplate)
], InputText);

export { INPUT_TEXT_ATTRIBUTES, InputText };
