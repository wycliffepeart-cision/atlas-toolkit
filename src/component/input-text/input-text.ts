import InputTextTemplate from './input-text.html';
import { Template } from '../../decorators/template';

export const INPUT_TEXT_ATTRIBUTES = {
  DISABLED: 'disabled',
  REQUIRED: 'required',
  READONLY: 'readonly',
  NAME: 'name',
  VALUE: 'value',
  MAX_LENGTH: 'maxlength',
  PATTERN: 'pattern',
  PLACEHOLDER: 'placeholder',
  STATE: 'state',
};

const INPUT_STATES = {
  ATLAS_INVALID: 'atlas--invalid',
};

@Template(InputTextTemplate)
export class InputText extends HTMLElement {
  static template;

  static get observedAttributes() {
    return Object.values(INPUT_TEXT_ATTRIBUTES);
  }

  attributeChangedCallback(name, _oldValue, newValue) {
    this.#attributePassthrought(name, newValue, [INPUT_TEXT_ATTRIBUTES.STATE]);
  }

  connectedCallback() {
    if (!this.isConnected) {
      return;
    }

    this.#inputElem.addEventListener('change', this.#handleInputChange);
    this.#inputElem.addEventListener('invalid', this.#handleInputInvalid);
  }

  disconnectedCallback() {
    this.#inputElem.removeEventListener('change', this.#handleInputChange);
    this.#inputElem.removeEventListener('invalid', this.#handleInputInvalid);
  }

  #attributePassthrought(name, value, exclude = []) {
    if (exclude.includes(name)) {
      return;
    }

    if (value === null) {
      this.#inputElem.removeAttribute(name);
    } else {
      this.#inputElem.setAttribute(name, value);
    }
  }

  #handleInputChange = ({ target }) => {
    this.dispatchEvent(new CustomEvent('change', { bubbles: true, composed: true }));

    const isValid = target.checkValidity();
    this.toggleAttribute(INPUT_STATES.ATLAS_INVALID, !isValid);
  };

  #handleInputInvalid = () => {
    this.dispatchEvent(new CustomEvent('invalid', { bubbles: true, composed: true }));
  };

  get #inputElem(): HTMLInputElement {
    return this.shadowRoot.getElementById('Input') as HTMLInputElement;
  }

  get value() {
    return this.#inputElem.value;
  }

  set value(value) {
    this.#inputElem.value = value;
  }

  get disabled() {
    const disabledAttr = this.getAttribute('disabled');
    return disabledAttr !== null && disabledAttr !== 'false';
  }

  set disabled(value) {
    this.toggleAttribute('disabled', value);
  }

  checkValidity() {
    const isValid = this.#inputElem.checkValidity();
    this.toggleAttribute(INPUT_STATES.ATLAS_INVALID, !isValid);
    return isValid;
  }

  get valid() {
    return this.#inputElem.validity.valid;
  }

  set valid(value) {
    this.#inputElem.setCustomValidity(value ? '' : 'invalid');
    this.#inputElem.checkValidity();

    this.toggleAttribute(INPUT_STATES.ATLAS_INVALID, !value);
  }
}

if (!customElements.get('atlas-input-text')) customElements.define('atlas-input-text', InputText);