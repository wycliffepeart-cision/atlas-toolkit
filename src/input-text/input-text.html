<template>
	<input id="Input" type="text" part="input" />

	<label for="Input" part="label-wrapper">
		<slot name="label" part="label"></slot>
		<slot name="required" part="required">*</slot>
	</label>

	<slot name="prefix" part="prefix"></slot>
	<slot name="suffix" part="suffix"></slot>

	<slot name="helperText" part="helper-text"></slot>
	<slot name="validationText" part="validation-text"></slot>
</template>
