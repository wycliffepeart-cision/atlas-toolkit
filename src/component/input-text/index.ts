import { InputText } from './input-text';

if (!customElements.get('atlas-input-text')) customElements.define('atlas-input-text', InputText);
