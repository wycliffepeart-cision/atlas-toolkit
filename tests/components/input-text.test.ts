import { fireEvent } from '@testing-library/dom';
import '../../src/component/input-text';

const INPUT_TEXT_TAG = 'atlas-input-text';
const MOCK_STRING = 'TEST TEXT VALUE';

afterEach(() => {
  const element = document.querySelector('body > *:first-child');
  element?.remove();
});

describe('Input', () => {
  it('construts', async () => {
    await customElements.whenDefined(INPUT_TEXT_TAG);
    const lement = document.createElement(INPUT_TEXT_TAG);
    document.body.appendChild(lement);

    (expect(lement) as any).toBeInTheDocument();
  });

  it('matches snapshot', async () => {
    await customElements.whenDefined(INPUT_TEXT_TAG);
    const element = document.createElement(INPUT_TEXT_TAG);
    document.body.appendChild(element);

    (expect(element.shadowRoot.innerHTML) as any).toMatchSnapshot();
  });

  it('is matches snapshot when disabled', async () => {
    await customElements.whenDefined(INPUT_TEXT_TAG);
    const element = document.createElement(INPUT_TEXT_TAG);
    document.body.appendChild(element);

    element.toggleAttribute('disabled', true);

    (expect(element.shadowRoot.innerHTML) as any).toMatchSnapshot();
  });

  it('is matches snapshot when disabled toggled on/off', async () => {
    await customElements.whenDefined(INPUT_TEXT_TAG);
    const element = document.createElement(INPUT_TEXT_TAG);
    document.body.appendChild(element);

    element.toggleAttribute('disabled', true);
    element.toggleAttribute('disabled', false);

    (expect(element.shadowRoot.innerHTML) as any).toMatchSnapshot();
  });

  it('triggers onChange proxy event when updated', async () => {
    await customElements.whenDefined(INPUT_TEXT_TAG);
    const element = document.createElement(INPUT_TEXT_TAG);
    document.body.appendChild(element);

    const innerInput = element.shadowRoot.getElementById('Input');

    const mockChangeEvent = jest.fn();

    element.addEventListener('change', mockChangeEvent);
    fireEvent.change(innerInput, { target: { value: MOCK_STRING } });
    (expect(mockChangeEvent) as any).toHaveBeenCalledTimes(1);
  });

  it('triggers onInvalid proxy event checkValidity is called', () => {
    const element: any = document.createElement(INPUT_TEXT_TAG);
    document.body.appendChild(element);
    element.toggleAttribute('required', true);

    const mockChangeEvent = jest.fn();

    element.addEventListener('invalid', mockChangeEvent);
    element.checkValidity();

    (expect(mockChangeEvent) as any).toHaveBeenCalledTimes(1);
  });

  describe('Text JS API', () => {
    describe('.value', () => {
      it('supports .value getter', async () => {
        await customElements.whenDefined(INPUT_TEXT_TAG);
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);
        (expect(inputElem.value) as any).toEqual('');
      });

      it('supports .value setter', async () => {
        await customElements.whenDefined(INPUT_TEXT_TAG);
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);
        inputElem.value = MOCK_STRING;
        (expect(inputElem.value) as any).toEqual(MOCK_STRING);
      });

      it('supports .value from value attribute', async () => {
        await customElements.whenDefined(INPUT_TEXT_TAG);
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);
        inputElem.setAttribute('value', MOCK_STRING);
        (expect(inputElem.value) as any).toEqual(MOCK_STRING);
      });
    });

    describe('.disabled', () => {
      it('getter (default)', async () => {
        await customElements.whenDefined(INPUT_TEXT_TAG);
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);
        (expect(inputElem.disabled) as any).toEqual(false);
      });

      it('setter (identity)', async () => {
        await customElements.whenDefined(INPUT_TEXT_TAG);
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);
        inputElem.disabled = true;
        (expect(inputElem.disabled) as any).toEqual(true);
      });

      it('from toggle attribute true', async () => {
        await customElements.whenDefined(INPUT_TEXT_TAG);
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);
        inputElem.toggleAttribute('disabled', true);
        (expect(inputElem.disabled) as any).toEqual(true);
      });

      it('from non-toggle attribute false', async () => {
        await customElements.whenDefined(INPUT_TEXT_TAG);
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);
        inputElem.setAttribute('disabled', 'false');

        (expect(inputElem.disabled) as any).toEqual(false);
      });

      it('from non-toggle attribute', async () => {
        await customElements.whenDefined(INPUT_TEXT_TAG);
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);
        inputElem.setAttribute('disabled', MOCK_STRING);

        (expect(inputElem.disabled) as any).toEqual(true);
      });
    });

    describe('.valid', () => {
      it('getter (default)', () => {
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);

        (expect(inputElem.valid) as any).toEqual(true);
      });

      it('setter', () => {
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);

        inputElem.valid = false;
        (expect(inputElem.valid) as any).toEqual(false);
        (expect(inputElem.hasAttribute('atlas--invalid')) as any).toEqual(true);

        inputElem.valid = true;
        (expect(inputElem.valid) as any).toEqual(true);
      });
    });

    describe('.checkValidity', () => {
      it('verify validity', () => {
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);
        (expect(inputElem.checkValidity()) as any).toEqual(true);
      });

      it('Updates the custom atlas--invalid state', () => {
        const inputElem: any = document.createElement(INPUT_TEXT_TAG);
        inputElem.toggleAttribute('required', true);
        const innerInput = inputElem.shadowRoot.getElementById('Input');

        (expect(inputElem.checkValidity()) as any).toEqual(false);
        (expect(inputElem.hasAttribute('atlas--invalid')) as any).toEqual(true);
      });
    });
  });
});
