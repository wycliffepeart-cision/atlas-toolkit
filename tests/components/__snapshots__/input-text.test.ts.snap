// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input is matches snapshot when disabled 1`] = `
<input id="Input"
       type="text"
       part="input"
       disabled
>
<label for="Input"
       part="label-wrapper"
>
  <slot name="label"
        part="label"
  >
  </slot>
  <slot name="required"
        part="required"
  >
    *
  </slot>
</label>
<slot name="prefix"
      part="prefix"
>
</slot>
<slot name="suffix"
      part="suffix"
>
</slot>
<slot name="helperText"
      part="helper-text"
>
</slot>
<slot name="validationText"
      part="validation-text"
>
</slot>
`;

exports[`Input is matches snapshot when disabled toggled on/off 1`] = `
<input id="Input"
       type="text"
       part="input"
>
<label for="Input"
       part="label-wrapper"
>
  <slot name="label"
        part="label"
  >
  </slot>
  <slot name="required"
        part="required"
  >
    *
  </slot>
</label>
<slot name="prefix"
      part="prefix"
>
</slot>
<slot name="suffix"
      part="suffix"
>
</slot>
<slot name="helperText"
      part="helper-text"
>
</slot>
<slot name="validationText"
      part="validation-text"
>
</slot>
`;

exports[`Input matches snapshot 1`] = `
<input id="Input"
       type="text"
       part="input"
>
<label for="Input"
       part="label-wrapper"
>
  <slot name="label"
        part="label"
  >
  </slot>
  <slot name="required"
        part="required"
  >
    *
  </slot>
</label>
<slot name="prefix"
      part="prefix"
>
</slot>
<slot name="suffix"
      part="suffix"
>
</slot>
<slot name="helperText"
      part="helper-text"
>
</slot>
<slot name="validationText"
      part="validation-text"
>
</slot>
`;
